### ISSUE TITLE
MXNet engine call failed: MXNetError: Type inconsistent, Provided = int32,  inferred type = int64

### ISSUE THREAD

**Issue by kimim** [awslabs/djl#317]
## Description

If the parse labels with datatype other than int32 to TopKAccuracy.updateAccumulator,  there will be an exeption: 

```
   Unhandled ai.djl.engine.EngineException
   MXNet engine call failed: MXNetError: Type inconsistent, Provided = int32,
   inferred type = int64
```

### Expected Behavior
int64 and int32 should be accepted

### Error Message

```
1. Unhandled ai.djl.engine.EngineException
   MXNet engine call failed: MXNetError: Type inconsistent, Provided = int32,
   inferred type = int64


             JnaUtils.java: 1808  ai.djl.mxnet.jna.JnaUtils/checkCall
             JnaUtils.java:  502  ai.djl.mxnet.jna.JnaUtils/imperativeInvoke
         FunctionInfo.java:   91  ai.djl.mxnet.jna.FunctionInfo/invoke
          MxNDManager.java:  324  ai.djl.mxnet.engine.MxNDManager/invoke
            MxNDArray.java:  443  ai.djl.mxnet.engine.MxNDArray/eq
         TopKAccuracy.java:   97  ai.djl.training.evaluator.TopKAccuracy/lambda$accuracyHelper$0
          IntPipeline.java:  180  java.util.stream.IntPipeline$1$1/accept
              Streams.java:  104  java.util.stream.Streams$RangeIntSpliterator/forEachRemaining
          Spliterator.java:  699  java.util.Spliterator$OfInt/forEachRemaining
     AbstractPipeline.java:  484  java.util.stream.AbstractPipeline/copyInto
     AbstractPipeline.java:  474  java.util.stream.AbstractPipeline/wrapAndCopyInto
     AbstractPipeline.java:  550  java.util.stream.AbstractPipeline/evaluate
     AbstractPipeline.java:  260  java.util.stream.AbstractPipeline/evaluateToArrayNode
    ReferencePipeline.java:  517  java.util.stream.ReferencePipeline/toArray
         TopKAccuracy.java:  100  ai.djl.training.evaluator.TopKAccuracy/accuracyHelper
     AbstractAccuracy.java:   83  ai.djl.training.evaluator.AbstractAccuracy/updateAccumulator
```

## How to Reproduce?
invoke TopKAccuracy.updateAccumulator with label in datatype int64

## What have you tried to solve it?

1. modify this line: https://github.com/awslabs/djl/blob/1df24cd6231f89a7e4c79178255f178a87a131e1/api/src/main/java/ai/djl/training/evaluator/TopKAccuracy.java#L78

from 
```java
       NDArray topKPrediction = prediction.argSort(axis).toType(DataType.INT32, false);
```

to

```java
       NDArray topKPrediction = prediction.argSort(axis).toType(DataType.INT64, false);
```

2. modify this line: https://github.com/awslabs/djl/blob/1df24cd6231f89a7e4c79178255f178a87a131e1/api/src/main/java/ai/djl/training/evaluator/TopKAccuracy.java#L97

from

```java
                                                        .eq(label.flatten())
```

to

```java
                                                        .eq(label.flatten().toType(INT64, false)
```

**Comment by lanking520** [comment]
I think we can add an option to allow int64 input. But that should apply to both label and pred

**Comment by kimim** [comment]
PR #364 try to fix this issue.

**Comment by frankfliu** [comment]
Fixed by #364
